src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -100,6 +114,109 @@ pub struct RSAggregateResult {
     pub type_mask: RSResultTypeMask,
 }
 
+impl RSAggregateResult {
+    /// Create a new aggregate result with th given capacity
+    pub fn new(cap: usize) -> Self {
+        Self {
+            num_children: 0,
+            children_cap: cap as c_int,
+            children: Self::new_children(cap),
+            type_mask: RSResultTypeMask::empty(),
+        }
+    }
+
+    fn push(&mut self, result: &mut RSIndexResult) {
```

Comment:
Do you actually need  `&mut RsIndexResult` here?

You can cast `&RsIndexResult` to a `*mut` using `result as *const _ as *mut _;`. 
I know it's kind of hacky but afaik it's a common pattern which will save us from relying on `mut` in the Rust API.

Reply:
Nope, `mut` is not needed. Nice, trick!! Done in -> https://github.com/RediSearch/RediSearch/pull/6490/commits/788d99821c04dc989e1c346305f6687b0fb4bf7e